lib/wasix/src/state/func_env.rs

```diff
@@ -191,10 +191,17 @@ impl WasiFunctionEnv {
                     _ => 0,
                 }
             } else if let Some(data_end) = data_end {
-                match data_end.get(store) {
+                let data_end = match data_end.get(store) {
                     wasmer::Value::I32(a) => a as u64,
                     wasmer::Value::I64(a) => a as u64,
                     _ => 0,
+                };
+                // It's possible for the data section to be above the stack, we check for that here and
+                // if it is, we'll assume the stack starts at address 0
+                if data_end >= stack_base {
```

Comment:
Are we sure the stack actually starts at 0?
We check the LLVM configuration in the Rust targets.

Reply:
I'm fairly confident that that should be the case, considering:
* The top of the stack falls at directly 2^20 (e.g. 1MB) by default.
* I don't know of any other memory regions that would live below the stack?